dashboard/report/generate_json_data.py

```diff
@@ -0,0 +1,12 @@
+def generate_json_data(
```

Comment:
I'd suggest a more suitable name for function, e.g. `read_stages_stats`

Reply:
on top of that you can specify what exactly is in these params
`second_stage`, `third_stage` and `fifth_stage`, the names need to be a bit more descriptive